core/include/seqan/align/evaluate_alignment.h

```diff
@@ -252,6 +252,241 @@ TScoreVal computeAlignmentStats(Align<TGaps, TAlignSpec> const & align,
+        }
+        else
+        {
+            // Compute the alignment score and register in stats.
+            TAlphabet c0 = *it0, c1 = *it1;
+            if (c0 == c1)
+            {
+                matchString[pos] = matchChar;
+            }
+            else
+            {
+                matchString[pos] = mismatchChar;
+            }
+        }
+    }
+    SEQAN_ASSERT(it0 == itEnd0);
+    SEQAN_ASSERT(it1 == itEnd1);
+
+    return matchString;
+}
+
+template <typename TGaps>
+String<char> alignmentIdentityString(TGaps const & queryRow,
+                                     TGaps const & referenceRow)
+{
+    return alignmentIdentityString(queryRow, referenceRow, '|', '*', ' ');
+}
+
+template <typename TAlphabet, typename TAlignSpec>
+String<char> alignmentIdentityString(Align<TAlphabet, TAlignSpec> const & align,
+                                     char const matchChar,
+                                     char const mismatchChar,
+                                     char const gapChar)
+{
+    // Check that the supplied alignment is pairwise
+    SEQAN_ASSERT_EQ_MSG(length(rows(align)), 2u, "Only works with pairwise alignments.");
+
+    typedef Align<TAlphabet, TAlignSpec> const TAlign;
+    typedef typename Row<TAlign>::Type TGaps;
+
+    TGaps queryRow = row(align, 0);
+    TGaps referenceRow = row(align, 1);
+
+    return alignmentIdentityString(queryRow, referenceRow, matchChar, mismatchChar, gapChar);
+}
+
+template <typename TSource, typename TAlignSpec>
+String<char> alignmentIdentityString(Align<TSource, TAlignSpec> const & align)
+{
+    return alignmentIdentityString(align, '|', '*', ' ');
+}
+
+
+// ----------------------------------------------------------------------------
+// Function alignmentSimilarityString()
+// ----------------------------------------------------------------------------
+
+/*!
+ * @fn Align#alignmentSimilarityString()
+ * @headerfile <seqan/align.h>
+ * @brief Returns a CharString of the similarity of positions in an alignment
+ *
+ * @signature CharString computeAlignmentSimilarityString();
+ *
+ * @param align The @link Align @endlink object to score for similarity.
+ * @param score The @link Score @endlink object to use for the scoring scheme.
+ * @param match The character to use for similarity at match positions
+ * @param positive The character to use for similarity at positive-scoring non-match positions
+ * @param mismatch The character to use for similarity at negative-scoring non-match positions
+ * @param gap The character to use for similarity at gap positions
+ *
+ * @signature CharString A string of characters representing alignment position similarity
+ *
+ * @section Examples
+ *
+ * @include demos/align/compute_match_string.cpp
+ *
+ * The output is as follows:
+ *
+ * @include demos/align/compute_match_string.cpp.stdout
+ */
+
+template <typename TGaps, typename TScoreVal, typename TScoreSpec>
+String<char> alignmentSimilarityString(TGaps & queryRow,
```

Comment:
Now you can reuse the interface from above to do the same here.
Maybe recall this interface (above) to _transformAlignment(Gaps, Gaps, BinaryOperator).
This should be then an internal interface, that's why the underscore at the beginning.